model/map/src/main/java/org/keycloak/models/map/storage/tree/TreeStorageNodeInstance.java

```diff
@@ -0,0 +1,128 @@
+/*
+ * Copyright 2021 Red Hat, Inc. and/or its affiliates
+ * and other contributors as indicated by the @author tags.
+ * 
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ * 
+ * http://www.apache.org/licenses/LICENSE-2.0
+ * 
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.keycloak.models.map.storage.tree;
```

Comment:
I think not because of the following:
- Tree storage will only be a few classes
- Tree storage is based on map storage, and interferes with it
- In most cases, map storage will be used in tree setup